src/api/index.ts

```diff
@@ -0,0 +1,103 @@
+import { setProvider } from './contracts'
+import { promisedWeb3 } from './web3Provider'
+import { promisedTokens } from './Tokens'
+import { promisedDutchX } from './DutchX'
+
+import { TokenCode, Account, Balance } from 'types'
+import { dxAPI } from './types'
+
+const promisedAPI = initAPI()
+
+export const getCurrentAccount = async () => {
+  const { web3 } = await promisedAPI
+
+  return web3.getCurrentAccount()
+}
+
+export const getAllAccounts = async () => {
+  const { web3 } = await promisedAPI
+
+  return web3.getAccounts()
+}
+
+// ether balance, not ETH tokens
+export const getETHBalance = async () => {
+  const { web3 } = await promisedAPI
+
+  return web3.getETHBalance()
+}
+
+// ETH token balance
+export const getCurrentBalance = async (account?: Account) => {
+  const { Tokens, web3 } = await promisedAPI
+
+  // account would normally be taken from redux state and passed inside an action
+  // but just in case
+  if (!account) account = await web3.getCurrentAccount()
+
+  return Tokens.getTokenBalance('ETH', account)
+}
+
+// TODO: remove ['ETH', 'GNO'] default, use actions for this
+export const getTokenBalances = async (codeList: TokenCode[] = ['ETH', 'GNO'], account?: Account) => {
```

Comment:
why is this called codeList and not tokenList?

Reply:
don't know, will change to `tokenList`